wgengine/userspace.go

```diff
@@ -255,32 +255,27 @@ func newUserspaceEngineAdvanced(logf logger.Logf, tundev *tstun.TUN, routerGen R
 	return e, nil
 }
 
-// pinger sends ping packets for a few seconds.
-//
-// These generated packets are used to ensure we trigger the spray logic in
-// the magicsock package for NAT traversal.
-func (e *userspaceEngine) pinger(peerKey wgcfg.Key, ips []wgcfg.IP) {
-	e.logf("generating initial ping traffic to %s (%v)", peerKey.ShortString(), ips)
-	var srcIP packet.IP
+type pinger struct {
+	e      *userspaceEngine
+	done   chan struct{}
```

Comment:
add a comment to clarify that this isn't the ctx.Done() channel. Next to the context.CancelFunc, it's easy to assume it is.